lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -640,36 +648,42 @@ static void AddRequiredAliases(Block *block, lldb::StackFrameSP &stack_frame_sp,
     swift::BoundGenericEnumType *optional_type =
         referent_type->getAs<swift::BoundGenericEnumType>();
 
-    if (!optional_type)
-      return;
+    if (!optional_type || optional_type->getGenericArgs().empty())
+      return Status("Could not add required aliases: could not get optional "
+                    "type for self.");
 
     swift::Type first_arg_type = optional_type->getGenericArgs()[0];
 
-    swift::ClassType *self_class_type =
-        first_arg_type->getAs<swift::ClassType>();
-
-    if (!self_class_type)
-      return;
+    // Only class types can be weakly captured.
+    if (!llvm::isa<swift::ClassType>(first_arg_type) &&
+        !llvm::isa<swift::BoundGenericClassType>(first_arg_type))
+      return Status("Could not add required aliases: weakly captured type is "
```

Comment:
```suggestion
      return Status("Unable to add the aliases the expression needs because "
```

If the word "Unable" isn't a valid option, then I suggest the contraction "Couldn't".

Reply:
Since "weakly captured" is already part of the Swift lingo I think we should keep it, maybe:

"Unable to add the aliases the expression needs because weakly captured type is not a class type"?